ocaml/testsuite/tests/typing-layouts/annots.ml

```diff
@@ -0,0 +1,402 @@
+(* TEST
+   * expect
+   flags = "-extension layouts"
+*)
+
+
+type t_value : value
+type t_imm : immediate
+type t_imm64 : immediate64
+;;
+[%%expect{|
+type t_value : value
+type t_imm : immediate
+type t_imm64 : immediate64
+|}];;
+
+type t_any : any;;
+
+[%%expect{|
+failure
+|}]
+
+type t_void : void
+
+[%%expect{|
+failure
+|}]
+
+(***************************************)
+(* Test 1: annotation on type variable *)
+
+let x : int as ('a : value) = 5
+let x : int as ('a : immediate) = 5
+;;
+[%%expect {|
+val x : int = 5
+val x : int = 5
+|}]
+
+let x : int as ('a : any) = 5;;
+
+[%%expect{|
+extension
+|}]
+(* CR layouts: fix when [any] becomes available in [layouts] *)
+
+let x : (int as ('a : immediate)) list as ('b : value) = [3;4;5]
+;;
+[%%expect {|
+val x : int list = [3; 4; 5]
+|}]
+
+let x : int list as ('a : immediate) = [3;4;5]
+;;
+[%%expect {|
+Line 1, characters 8-30:
+1 | let x : (int list : immediate) = [3;4;5]
+            ^^^^^^^^^^^^^^^^^^^^^^
+Error: Bad layout annotation:
+         int list has layout value, which is not a sublayout of immediate.
+|}]
+
+(****************************************)
+(* Test 2: Annotation on type parameter *)
+
+type ('a : immediate) t2_imm
+type (_ : immediate) t2_imm'
+type t = int t2_imm
+type t = bool t2_imm
+;;
+[%%expect {|
+type ('a : immediate) t2_imm
+type (_ : immediate) t2_imm'
+type t = int t2_imm
+type t = bool t2_imm
+|}]
+
+module M1 : sig
+  type ('a : immediate) t
+end = struct
+  type (_ : immediate) t
+end
+
+module M2 : sig
+  type (_ : immediate) t
+end = struct
+  type ('a : immediate) t
+end
+
+[%%expect {|
+success
```

Comment:
Why isn't this the text of the success?  (This hold below, I'm just not going to repeat it)